system-reinstall-bootc/src/prompt.rs

```diff
@@ -92,6 +93,42 @@ pub(crate) fn ask_yes_no(prompt: &str, default: bool) -> Result<bool> {
         .context("prompting")
 }
 
+pub(crate) fn mount_warning() -> Result<()> {
+    let mounts = bootc_mount::run_findmnt(&[], None)?;
+
+    let problematic_filesystems: Vec<&Filesystem> = mounts
+        .filesystems
+        .iter()
+        .find(|mount| mount.target == "/")
+        .map(|root| {
+            root.children
+                .iter()
+                .flatten()
+                .filter(|child| child.source == root.source)
```

Comment:
So...this seems like it will basically only find btrfs subvolumes, right? I think that's the only real sane case where the same source device can be mounted multiple times in that way.

In Linux one *can* mount e.g. the same block device multiple times but for filesystems without subvolumes there's not much use case for it.

Reply:
So for someone who has e.g. `/home` or `/var/log` as a LVM logical volume, the sources are different so this check won't trigger.

However, the hard part here is to generalize this, we're going to need to exclude e.g. the [API filesystems](https://www.freedesktop.org/wiki/Software/systemd/APIFileSystems/) among many other things. I'm sure there's code to do this, but we'd need to do some investigation.